fvm/Grid/PXiGrid/XiBiUniformThetaGridGenerator.cpp

```diff
@@ -77,12 +76,13 @@ bool XiBiUniformThetaGridGenerator::Rebuild(const real_t, const len_t, MomentumG
     for (len_t i = this->nxiSep; i < this->nxi; i++)
         dth[i] = dth_up;
 	
-	th_f[0] = thMin;
-	xi_f[this->nxi] = cos(thMin);
+    th_f[0] = thMin;
+    xi_f[this->nxi] = cos(thMin);
+    xi_f[0] = cos(thMax);
    
-    for (len_t i = 1; i <= this->nxi; i++){
-    	th_f[i] = th_f[i-1] + dth[i-1];
-    	xi_f[this->nxi-i] = cos(th_f[i]);
+    for (len_t i = 1; i <= this->nxi-1; i++){
+        th_f[i] = th_f[i-1] + dth[i-1];
+    	  xi_f[this->nxi-i] = cos(th_f[i]);
         }  
```

Comment:
These two lines still have a weird indentation (the curly bracket should be aligned with the start of ``for`` on line 83